### PR TITLE
Shared library: remove poco dependency authored by daohu527

### PR DESCRIPTION
**Pull request by storypku**
With original notes/license from the Poco project.
And test cases for shared_library specifically.

@daohu527

### CODE REVIEW

**Comment by daohu527** [comment]
Test cases and submissions looks good.

**Comment by storypku** [comment]
> Test cases and submissions looks good.

We can continue to check whether calling `unload()` inside dctor is OK from the whole cyber/class_loader submodule after this PR.

**Review by jinghaomiao — APPROVED** [review]

File: cyber/class_loader/BUILD [hunk 1]
```diff
@@ -19,7 +19,7 @@ cc_library(
     deps = [
         "//cyber:init",
         "//cyber/common:log",
-        "@poco//:PocoFoundation",
+        "//cyber/class_loader/shared_library",
```
Comment (xiaoxq) [inline cyber/class_loader/BUILD:22]: buildifier

File: cyber/class_loader/shared_library/exceptions.h [hunk 2]
```diff
@@ -0,0 +1,43 @@
+/******************************************************************************
+ * Copyright 2020 The Apollo Authors. All Rights Reserved.
```
Comment (xiaoxq) [inline cyber/class_loader/shared_library/exceptions.h:2]: 2021

**Review by xiaoxq — APPROVED** [review]

File: cyber/class_loader/shared_library/exceptions.h (hunk 2 above)
Comment (storypku) [reply cyber/class_loader/shared_library/exceptions.h:2]: Hahaha, forgot new year has come.

**Pull request merged**